src/test/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializerTest.java

```diff
@@ -98,4 +117,24 @@ void testRandomTxPower() throws Exception {
 			assertTrue(txPower >= TPC.MIN_TX_POWER && txPower <= TPC.MAX_TX_POWER);
 		}
 	}
+
+	@Test
+	@Order(3)
+	void testRandomTxPowerPerAp() throws Exception {
+		TPC optimizer = new RandomTxPowerInitializer(
+			createModel(), TEST_ZONE, createDeviceDataManager(), true, new Random(0)
+		);
+		Map<String, Map<String, Integer>> txPowerMap =
+			optimizer.computeTxPowerMap();
+		Set<Integer> txPowerSet = new TreeSet<>();
+		for (String serialNumber : txPowerMap.keySet()) {
+			txPowerSet.addAll(txPowerMap.get(serialNumber).values());
+		}
+
+		// different values per AP
+		assertNotEquals(1, txPowerSet.size());
```

Comment:
@elludraon do you think this test should be more specific than just testing that the set has more than one element?

Reply:
This should be `assertTrue(txPowerSet.size() > 1)` because you don't want to pass on size=0.